src/librustc_typeck/check/demand.rs

```diff
@@ -328,8 +328,7 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 // (But, also check check the `expn_info()` to see if this is
                 // a macro; if so, it's hard to extract the text and make a good
                 // suggestion, so don't bother.)
-                if self.infcx.can_sub(self.param_env, checked, &expected).is_ok() &&
-                   sp.ctxt().outer().expn_info().is_none() {
```

Comment:
(maybe we should make it assert?)

Reply:
No, I think it'd be better to revert that last commit. The checks are in different spans and we should only suggest if _both_ spans are outside of macros. Originally I thought that it would make sense to suggest if the macro was local (meaning that the user could do something about it), but in that case the span already points at the problematic place, and I will assume that anyone using macros would have enough knowledge of the language to deal with an E0308 between `u32` and `&{integer}`. What do you think?